### DISCUSSION TITLE
Multicall aggregate analytics

### DISCUSSION THREAD

**Discussion by hammeiam** [wevm/viem#400] [Question] [answered]
Is there any way to debug or otherwise analyze the amount of calls that are being bundled into multicall calls, short of inspected my RPC provider logs? I'd love to do a before/after comparison. I'd also love to validate on my own that both `contract.read` and `publicClient.readContract` calls get bundled the same way. I could also see this helping people to pick an optimized `multicall.wait` value.

**Comment by jxom** [comment, marked as answer]
Eventually we could have `onRequest`/`onResponse` callbacks that you can pass to the viem Transports to hook in for analysis.

**Reply by fubhy** [reply]
I would very much like to see that for both performance metrics, analytics and generic tracing support: https://github.com/wagmi-dev/viem/discussions/261

Definitely also for calls, but especially for sending transactions, understanding where there are bottlenecks, frequent errors / error clusters is crucial for building good UXs on top of such a dynamic, stateful environment as the blockchain.
